### ISSUE TITLE
Update module github.com/vmware/govmomi to v0.37.0 causes build failures

### ISSUE THREAD

**Issue by codeboten** [open-telemetry/opentelemetry-collector-contrib#32453]
### Component(s)

receiver/vcenter

### Describe the issue you're reporting

See https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32436

```
=== Failed
=== FAIL: . TestScrape (0.01s)
    scraper_test.go:58: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/vcenterreceiver/scraper_test.go:58
        	            				/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/vcenterreceiver/scraper_test.go:33
        	Error:      	Received unexpected error:
        	            	unable to get datacenter lists: POST "/sdk": 404 Not Found
        	Test:       	TestScrape

=== FAIL: . TestScrape_TLS (0.02s)
    scraper_test.go:58: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/vcenterreceiver/scraper_test.go:58
make[2]: *** [../../Makefile.Common:126: test] Error 1
make[1]: *** [Makefile:165: receiver/vcenterreceiver] Error 2
make: *** [Makefile:113: gotest] Error 2
        	            				/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/vcenterreceiver/scraper_test.go:51
        	Error:      	Received unexpected error:
        	            	unable to get datacenter lists: POST "/sdk": 404 Not Found
        	Test:       	TestScrape_TLS

=== FAIL: . TestScrape (re-run 1) (0.01s)
    scraper_test.go:58: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/vcenterreceiver/scraper_test.go:58
        	            				/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/vcenterreceiver/scraper_test.go:33
        	Error:      	Received unexpected error:
        	            	unable to get datacenter lists: POST "/sdk": 404 Not Found
        	Test:       	TestScrape

=== FAIL: . TestScrape_TLS (re-run 1) (0.02s)
    scraper_test.go:58: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/vcenterreceiver/scraper_test.go:58
        	            				/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/vcenterreceiver/scraper_test.go:51
        	Error:      	Received unexpected error:
        	            	unable to get datacenter lists: POST "/sdk": 404 Not Found
        	Test:       	TestScrape_TLS

```

**Comment by djaglowski** [comment]
@schmikei @StefanKurek, can either of you please take this? This is blocking other PRs

**Comment by djaglowski** [comment]
Thanks @StefanKurek!

**Comment by StefanKurek** [comment]
@djaglowski So I "think" I at least have some idea where to start looking. So at the very least, with the new `govmomi` update, I know that [this](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/98b4c27a1563e53ae152150119f9ad5c06d4149b/receiver/vcenterreceiver/client.go#L74) is now using a new type of call [RetrievePropertiesEx](https://vdc-repo.vmware.com/vmwb-repository/dcr-public/e1580753-930f-4c00-9105-77dd2e8913f8/98ce8e90-a51b-4e41-8dac-3e8fd5409f2b/vmodl.query.PropertyCollector.html#retrievePropertiesEx) as the old one `RetrieveProperties` was deprecated as of version 4.1 of the API.

This results in the mock server not returning a correct response to seed the finder with a default `datacenter`, and then some (all?) consecutive client calls will fail (as they are expecting this).

Also, "all" the mock responses may need to be updated to `RetrievePropertiesExResponse` types rather than just for this one call. I'm not completely sure of this though, and would require more digging (would also probably become obvious after fixing that initial mocked response).

I don't have time to work on this immediately. It may be a few days, but I will try to come back to it soon.

**Comment by StefanKurek** [comment]
We should be able to close this now

**Comment by djaglowski** [comment]
Resolved by #33721
